executor/src/witgen/global_constraints.rs

```diff
@@ -389,7 +389,7 @@ mod test {
 
     #[test]
     fn zero_one_two_three() {
-        let fixed = [0, 1, 2, 3, 0].map(|v| v.into());
+        let fixed = [0, 1, 2, 3].map(|v| v.into());
```

Comment:
This test wouldn't have worked before and is working now. It is exactly testing the case which is the motivation for this PR.